details/caching.md

```diff
@@ -0,0 +1,190 @@
+  
+The second information only can be cached by the following rules:
+* `ldap_authorization`
+* `groups_provider_authorization`
+
+Both information can be cached by:
+* `ldap_auth` rule
+
+### Group caching
+
+Group caching looks simple. In general, it is - we get the user's groups from e.g. LDAP and store the information in the cache. But with server-side group filtering* enabled, it's not so simple. Why is that? Let's look at the following example:
+
+```yaml
+# Example: LDAP auth rules used in 3 blocks with cache configured at LDAP connector level (service definition level) and the LDAP has configured the server-side group filtering
+
+readonlyrest:
+
+  access_control_rules:  
+  - name: "Block 1"
+    [...]
+    ldap_auth:
+      name: ldap1
+      groups: [group1_*]
+
+  - name: "Block 2"
+    [...]
+    ldap_auth:
+      name: ldap1
+      groups: [group2_*]
+
+  - name: "Block 3"
+    [...]
+    ldap_auth:
+      name: ldap1
+      groups: [group3_*]
+
+  ldaps:
+  - name: ldap1
+    [...]
+    sever_side_groups_filtering: true
+    cache_ttl: 60 sec # service client definition level cache
+```
+
+This example is a slightly modified version of the previous ones:
+* we use the `ldap_auth` rule (authentication + authorization) instead of the `ldap_authentication` rule:
+  * the block `Block 1` matches only when the user can be authenticated in LDAP and they have groups that match `group1_*` wildcard
+  * the block `Block 2` matches only when the user can be authenticated in LDAP and they have groups that match `group2_*` wildcard
+  * the block `Block 3` matches only when the user can be authenticated in LDAP and they have groups that match `group3_*` wildcard
+* the LDAP connector "ldap1" has the server-side group filtering feature enabled
+
+Let's call `GET /my_index/_search` and see what happens. We see 3 calls to LDAP. Why is that? Because the service level cache was configured?! Yes, but this is because of the server-side group filtering. Let's describe what each block LDAP group call looked like:
+
+* the block `Block 1` - "find all groups for the user that start with `group1_`"
+* the block `Block 2` - "find all groups for the user that start with `group2_`"
+* the block `Block 3` - "find all groups for the user that start with `group3_`"
+
+Oh yes. It should be obvious now. We don't get all user groups, just a subset of all existing groups. We do cache the information, but each block can't find the information it's looking for in the cache. So we call LDAP in each block. 
+
+Is it bad or not? It depends. On the business case, of course. In some cases it's better to enable server-side rendering (e.g. when the LDAP groups are a really long list), but in other cases it's better to reduce LDAP calls (so more data can be sent over the network). In ROR, server-side group filtering is disabled by default. You should consider whether it should be enabled in your case. But be aware of the caching implications described above.
```

Comment:
Maybe let's hilight this consideration about when to use server side filtering in its own paragraph? Like "When to enable server-side groups filtering?"